content/ui-ux.md

```diff
@@ -0,0 +1,514 @@
+
+## User Interface Patterns
+
+There are some patterns for building user interfaces that are useful to keep in mind for a Meteor application.
+
+### Responsive Design
+### Accessibility
+
+XXX: Honestly I'm really not sure there's very much Meteor specific in these two headings. I'm not convinced they belong in the guide
+
+
+### Internationalization
+
+Internationalization (i18n) is the process of generalizing the UI of your app in such a way that it's easy to render all text in a different language. In Meteor, the excellent `tap:i18n` package provides an API for building translations and using them in your templates and elsewhere.
+
+#### Using `tap:i18n`
+
+XXX: I'm going to leave this section for now as it's kind of dependent on modules (it's more complex than maybe it needs to be in Todos right now due to all-packages).
+
+#### Places to translate
+
+It's useful to consider the various places in the system that user-readable strings can come out of and make sure that you are properly using the i18n system to generate those strings in each case.
+
+The obvious place for such strings is HTML -- the content of components that the user sees. Additionally any alerts or other messages that are generated on the client side should be translated also.
+
+Another place to be aware of is messages that are generated by the server that are user visible. An obvious place is emails and any other server generated messages (such as mobile push notifications), but more subtle places are return values and error messages on method calls (which should be sent over the wire in a generic form and translated on the client, see above).
+
+A final place where you may want to translate is the actual data in your database. Of course how you might go about this would be very much a problem unique to your application!
+
+### Event handling
+
+A large part of your UI involves responding to events initated by the user, and there are some steps that you should take to make sure your application does this well
+
+#### Throttling method calls on user action
+
+It's typical to make some kind of change to the database when a user takes an action. However it's important to make sure you don't do this *too* much. For instance, if you wish to save the user's text as they type in a textbox, you should take steps to make sure that you don't send method calls to your database more than every few hundred milliseconds.
+
+If you do not, you'll see performance problems across the board; you'll be flooding the user's network connection with a lot of minute changes, the UI will update on every keystroke, potentially causing poor performance, and your database will suffer with a lot of writes.
+
+To throttle writes, a typical approach is to use underscore's [`.throttle()`](http://underscorejs.org/#throttle) or [`.debounce()`](http://underscorejs.org/#debounce) functions. For instance, in the Todos example app, we throttle writes on user input to 300ms:
```

Comment:
Do we want to recommend one, or specify which is better? I bet a lot of people won't really get the difference between the two. I feel like this is a relatively important concept.

Also, maybe this would go in a "performance" guide?